File: schemaorg-rs/src/order.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/paymentDueDate>
///<https://schema.org/Date>
///<https://schema.org/DateTime>
pub type OrderPaymentDueDateFieldEnum = String;
///<https://schema.org/orderDate>
///<https://schema.org/Date>
///<https://schema.org/DateTime>
pub type OrderOrderDateFieldEnum = String;
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type OrderAdditionalTypeFieldEnum = String;
///<https://schema.org/Order>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct Order {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/paymentMethod>
    #[serde(rename = "paymentMethod")]
    #[serde_as(as = "OneOrMany<_>")]
    pub payment_method: Vec<OrderPaymentMethodFieldEnum>,
    ///<https://schema.org/billingAddress>
    #[serde(rename = "billingAddress")]
    #[serde_as(as = "OneOrMany<_>")]
    pub billing_address: Vec<PostalAddress>,
    ///<https://schema.org/broker>
    #[serde(rename = "broker")]
    #[serde_as(as = "OneOrMany<_>")]
    pub broker: Vec<OrderBrokerFieldEnum>,
    ///<https://schema.org/merchant>
    #[serde(rename = "merchant")]
    #[serde_as(as = "OneOrMany<_>")]
    pub merchant: Vec<OrderMerchantFieldEnum>,
    ///<https://schema.org/paymentDueDate>
    #[serde(rename = "paymentDueDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub payment_due_date: Vec<OrderPaymentDueDateFieldEnum>,
    ///<https://schema.org/seller>
    #[serde(rename = "seller")]
    #[serde_as(as = "OneOrMany<_>")]
    pub seller: Vec<OrderSellerFieldEnum>,
    ///<https://schema.org/isGift>
    #[serde(rename = "isGift")]
    #[serde_as(as = "OneOrMany<_>")]
    pub is_gift: Vec<String>,
    ///<https://schema.org/orderStatus>
    #[serde(rename = "orderStatus")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_status: Vec<OrderStatusEnum>,
    ///<https://schema.org/acceptedOffer>
    #[serde(rename = "acceptedOffer")]
    #[serde_as(as = "OneOrMany<_>")]
    pub accepted_offer: Vec<Offer>,
    ///<https://schema.org/orderNumber>
    #[serde(rename = "orderNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_number: Vec<String>,
    ///<https://schema.org/orderDate>
    #[serde(rename = "orderDate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_date: Vec<OrderOrderDateFieldEnum>,
    ///<https://schema.org/customer>
    #[serde(rename = "customer")]
    #[serde_as(as = "OneOrMany<_>")]
    pub customer: Vec<OrderCustomerFieldEnum>,
    ///<https://schema.org/paymentMethodId>
    #[serde(rename = "paymentMethodId")]
    #[serde_as(as = "OneOrMany<_>")]
    pub payment_method_id: Vec<String>,
    ///<https://schema.org/orderDelivery>
    #[serde(rename = "orderDelivery")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_delivery: Vec<ParcelDelivery>,
    ///<https://schema.org/discountCode>
    #[serde(rename = "discountCode")]
    #[serde_as(as = "OneOrMany<_>")]
    pub discount_code: Vec<String>,
    ///<https://schema.org/discountCurrency>
    #[serde(rename = "discountCurrency")]
    #[serde_as(as = "OneOrMany<_>")]
    pub discount_currency: Vec<String>,
    ///<https://schema.org/partOfInvoice>
    #[serde(rename = "partOfInvoice")]
    #[serde_as(as = "OneOrMany<_>")]
    pub part_of_invoice: Vec<Invoice>,
    ///<https://schema.org/orderedItem>
    #[serde(rename = "orderedItem")]
    #[serde_as(as = "OneOrMany<_>")]
    pub ordered_item: Vec<OrderOrderedItemFieldEnum>,
    ///<https://schema.org/paymentDue>
    #[serde(rename = "paymentDue")]
    #[serde_as(as = "OneOrMany<_>")]
    pub payment_due: Vec<String>,
    ///<https://schema.org/discount>
    #[serde(rename = "discount")]
    #[serde_as(as = "OneOrMany<_>")]
    pub discount: Vec<OrderDiscountFieldEnum>,
    ///<https://schema.org/confirmationNumber>
    #[serde(rename = "confirmationNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub confirmation_number: Vec<String>,
    ///<https://schema.org/paymentUrl>
    #[serde(rename = "paymentUrl")]
    #[serde_as(as = "OneOrMany<_>")]
    pub payment_url: Vec<String>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OrderMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OrderIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OrderDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OrderSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OrderAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OrderImageFieldEnum>,
}

File: schemaorg-ts/src/order.ts
import type * as all from "./index";

/**
 * <https://schema.org/paymentDueDate>
 **/
export type OrderPaymentDueDateFieldEnum = string;

/**
 * <https://schema.org/orderDate>
 **/
export type OrderOrderDateFieldEnum = string;

/**
 * <https://schema.org/additionalType>
 **/
export type OrderAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/Order>
 **/
export interface Order {
  "@context": string;
  "@type": "Order";

  /**
   * <https://schema.org/paymentMethod>
   **/
  paymentMethod: all.OneOrMany<all.OrderPaymentMethodFieldEnum>;

  /**
   * <https://schema.org/billingAddress>
   **/
  billingAddress: all.OneOrMany<all.PostalAddress>;

  /**
   * <https://schema.org/broker>
   **/
  broker: all.OneOrMany<all.OrderBrokerFieldEnum>;

  /**
   * <https://schema.org/merchant>
   **/
  merchant: all.OneOrMany<all.OrderMerchantFieldEnum>;

  /**
   * <https://schema.org/paymentDueDate>
   **/
  paymentDueDate: all.OneOrMany<all.OrderPaymentDueDateFieldEnum>;

  /**
   * <https://schema.org/seller>
   **/
  seller: all.OneOrMany<all.OrderSellerFieldEnum>;

  /**
   * <https://schema.org/isGift>
   **/
  isGift: all.OneOrMany<string>;

  /**
   * <https://schema.org/orderStatus>
   **/
  orderStatus: all.OneOrMany<all.OrderStatusEnum>;

  /**
   * <https://schema.org/acceptedOffer>
   **/
  acceptedOffer: all.OneOrMany<all.Offer>;

  /**
   * <https://schema.org/orderNumber>
   **/
  orderNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/orderDate>
   **/
  orderDate: all.OneOrMany<all.OrderOrderDateFieldEnum>;

  /**
   * <https://schema.org/customer>
   **/
  customer: all.OneOrMany<all.OrderCustomerFieldEnum>;

  /**
   * <https://schema.org/paymentMethodId>
   **/
  paymentMethodId: all.OneOrMany<string>;

  /**
   * <https://schema.org/orderDelivery>
   **/
  orderDelivery: all.OneOrMany<all.ParcelDelivery>;

  /**
   * <https://schema.org/discountCode>
   **/
  discountCode: all.OneOrMany<string>;

  /**
   * <https://schema.org/discountCurrency>
   **/
  discountCurrency: all.OneOrMany<string>;

  /**
   * <https://schema.org/partOfInvoice>
   **/
  partOfInvoice: all.OneOrMany<all.Invoice>;

  /**
   * <https://schema.org/orderedItem>
   **/
  orderedItem: all.OneOrMany<all.OrderOrderedItemFieldEnum>;

  /**
   * <https://schema.org/paymentDue>
   **/
  paymentDue: all.OneOrMany<string>;

  /**
   * <https://schema.org/discount>
   **/
  discount: all.OneOrMany<all.OrderDiscountFieldEnum>;

  /**
   * <https://schema.org/confirmationNumber>
   **/
  confirmationNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/paymentUrl>
   **/
  paymentUrl: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OrderMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OrderIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OrderDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OrderSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OrderAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OrderImageFieldEnum>;
}
